packages/vite/src/node/plugins/css.ts

```diff
@@ -1106,36 +1106,28 @@ async function minifyCSS(css: string, config: ResolvedConfig) {
   return code
 }
 
-// #1845
-// CSS @import can only appear at top of the file. We need to hoist all @import
-// to top when multiple files are concatenated.
-// #6333
-// CSS @charset must be the top-first in the file, hoist to top too
-async function hoistAtRules(css: string) {
-  const postcss = await import('postcss')
-  return (await postcss.default([AtRuleHoistPlugin]).process(css)).css
-}
-
-const AtRuleHoistPlugin: PostCSS.PluginCreator<any> = () => {
-  return {
-    postcssPlugin: 'vite-hoist-at-rules',
-    Once(root) {
-      const imports: PostCSS.AtRule[] = []
-      let charset: PostCSS.AtRule | undefined
-      root.walkAtRules((rule) => {
-        if (rule.name === 'import') {
-          // record in reverse so that can simply prepend to preserve order
-          imports.unshift(rule)
-        } else if (!charset && rule.name === 'charset') {
-          charset = rule
-        }
-      })
-      imports.forEach((i) => root.prepend(i))
-      if (charset) root.prepend(charset)
+export async function hoistAtRules(css: string) {
+  const s = new MagicString(css)
+  // #1845
+  // CSS @import can only appear at top of the file. We need to hoist all @import
+  // to top when multiple files are concatenated.
+  // match until semicolon that's not in quotes
+  s.replace(/@import\s*(?:"[^"]*"|'[^']*'|[^;]*).*?;/gm, (match) => {
```

Comment:
Those are real edge cases 🥲  It looks tricky to use `emptyString()` with `MagicString.replace` though, and doing so would make another copy of the string, which brings in some memory overhead. I feel like we could ignore them for now since there are unlikely to be instances of this in practice.

Otherwise, it may be simpler to go back to the postcss way 🤔

Reply:
There should be no such use method for this edge case list, but if you want to avoid it, you can use the following code. 😊

```ts
s.replace(/@import\s*(?:"[^"]*"|'[^']*'|[^;]*).*?;|\/\*(.|[\r\n])*?\*\/|\/\/.*/gm, (match) => {
  if (match[0] === '/') return match
  ...
})
```